### ISSUE TITLE
complex network accuracy on wifi

### ISSUE THREAD

**Issue by pelavoie** [metehancekic/wireless-fingerprinting#4]
Hi,
I cannot reproduce the complex network accuracy of 99.62% on the clean WiFi dataset reported in Table I. 
With the code from this repository or in a custom training harness, I get performance equivalent or worse than the "Real 2x" architecture (<98%). Is there something I am missing?

**Comment by metehancekic** [comment]
Hi,
Which dataset are you using, and do you use our code to train models (that is to say: same optimizer and scheduler). Since we are not allowed, we are not able to share the original clean WiFi dataset but simulated one.

**Comment by pelavoie** [comment]
I am using the dataset referred to in `data/data_readme.txt`.
Now I see that it says "The simulation based dataset [...]".
So this is a simulated dataset? How was it generated?

**Comment by metehancekic** [comment]
It is simulated based on the I-Q imbalance and Power amplifier nonlinearity models (Only these two). You can check the further details through our paper (section 4E):

https://arxiv.org/pdf/2002.10791.pdf

**Comment by metehancekic** [comment]
I will check with the simulated dataset and let you know about the performance.

**Comment by pelavoie** [comment]
Thanks.
To answer the 2nd question, yes I am using your code to train models, same optimizer and hyperparameters.

**Comment by metehancekic** [comment]
Hi, thanks for your patience. These are the results for the simulated dataset:

Real: 97.21
Real sqrt2x: 96.37
Real 2x: 96.53
Complex: 98.05

You can find the checkpoints to reproduce these results inside the repo (just pushed now).

As we said in the paper "We use the same CNN and training hyperparameters as before, except for the number of epochs which we set to 100".

The simulated dataset is only used for tables 7 and 9 in the arxiv paper.

**Comment by pelavoie** [comment]
Hi,
Thanks for verifying the results.

I verified the uploaded checkpoints and reproduce the same results.

I went back and trained using the simulated dataset with the default command (i.e.: `python cfo_channel_training_simulations.py`) on a fresh checkout of this repository.

I still get better test accuracy than the checkpoints with the real models (reim, reimsqrt2x, reim2x) and inferior results with the complex model (modrelu).

Here are the test accuracy averages of 5 training runs:

reim: 98.23
reimsqrt2x: 97.62
reim2x: 98.91
modrelu: 96.09

I followed the readme to setup the environment and double-checked the package versions.
I trained on CPU and GPU and obtained similar results.

The only difference I could find is that on GPU I have more recent CUDA and cuDNN versions.
> CUDA                    10.2.89\
> cuDNN                     7.6.5

Do you have any idea on what could be causing such different results?
